File: stock_service_consumer/src/main/java/com/github/marceloedudev/stockserviceconsumer/domain/sagas/SagaStep.java
package com.github.marceloedudev.stockserviceconsumer.domain.sagas;

import com.github.marceloedudev.stockserviceconsumer.packages.queue.MessageHandler;

import java.util.Optional;

public interface SagaStep {
    void setInvocation(MessageHandler handler);
    void setCompensation(MessageHandler handler);
    String getDescription();
    Optional<MessageHandler> getInvocation();
    Optional<MessageHandler> getCompensation();
}

File: auth_service/src/domain/sagas/SagaStep.ts
import MessageHandler from "@/packages/eventBus/abstractions/MessageHandler";

export default interface SagaStep {
    setInvocation(handler: MessageHandler): void;
    setCompensation(handler: MessageHandler): void;
    getDescription(): string;
    getInvocation(): MessageHandler | null;
    getCompensation(): MessageHandler | null;
}
